src/transcoding/codec/profile_video_class.c

```diff
@@ -63,11 +73,38 @@ tvh_codec_profile_video_setup(TVHVideoCodecProfile *self, tvh_ssc_t *ssc)
 {
     self->size.den = ssc->es_height;
     self->size.num = ssc->es_width;
-    if (self->height) {
-        self->size.den = self->height;
-        self->size.den += self->size.den & 1;
-        self->size.num = self->size.den * ((double)ssc->es_width / ssc->es_height);
-        self->size.num += self->size.num & 1;
+    switch (self->scaling_mode) {
+        case 0:
+            // scaling up and down
+            if (self->height) {
+                self->size.den = self->height;
+                self->size.den += self->size.den & 1;
+                self->size.num = (int)((double)self->size.den * ((double)ssc->es_width / (double)ssc->es_height));
+                self->size.num += self->size.num & 1;
+            }
+            break;
+        case 1:
+            // scaling up (only)
+            if ((self->height > 0) && (self->size.den < self->height)) {
+                self->size.den = self->height;
+                self->size.den += self->size.den & 1;
+                self->size.num = (int)((double)self->size.den * ((double)ssc->es_width / (double)ssc->es_height));
+                self->size.num += self->size.num & 1;
+            }
+            break;
+        case 2:
+            // scaling down (only)
+            if ((self->height > 0) && (self->size.den > self->height)) {
+                self->size.den = self->height;
+                self->size.den += self->size.den & 1;
+                self->size.num = (int)((double)self->size.den * ((double)ssc->es_width / (double)ssc->es_height));
+                self->size.num += self->size.num & 1;
+            }
+            break;
+        default:
+            // should never reach this point.
+            tvherror(LS_PROFILE, "self->scaling_mode unexpected value: valid range [0,1,2] received %d", self->scaling_mode);
```

Comment:
This should be something like `Invalid scaling mode: valid range 0-2, received %d`

Reply:
I tend to disagree .... let me explain:
- the user will not see this message unless someone changes the source code in order to add more options in the combo box. 
- this message is intended for developers.
- I can change it ... but frankly I think is correct as it is.

Reply:
Another option is to remove default: completely.

Reply:
An error message can be visual for a user and even for developers it should be clear to understand.

However, if there is no way that message will be shown normally it should be removed completely.